tensorflow_io/arrow/python/kernel_tests/arrow_test.py

```diff
@@ -0,0 +1,278 @@
+# License for the specific language governing permissions and limitations under
+# the License.
+# ==============================================================================
+"""Tests for ArrowDataset."""
+
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+
+from collections import namedtuple
+import os
+import socket
+import tempfile
+import threading
+import unittest
+
+_pyarrow_requirement_message = None
+try:
+  import pyarrow as pa
+  from pyarrow.feather import write_feather
+except ImportError as e:
+  _pyarrow_requirement_message = str(e)
+_have_pyarrow = _pyarrow_requirement_message is None
+
+from tensorflow_io.arrow.python.ops import arrow_dataset_ops
+from tensorflow.python.framework import dtypes
+from tensorflow.python.framework import tensor_shape
+from tensorflow.python.platform import test
+
+
+TestData = namedtuple("TestData", ["data", "output_types", "output_shapes"])
+
+
+class ArrowDatasetTest(test.TestCase):
+
+  @classmethod
+  def setUpClass(cls):
+
+    cls.scalar_data = [
+        [1, 2, -3, 4],
+        [1, 2, -3, 4],
+        [1, 2, -3, 4],
+        [1, 2, -3, 4],
+        [1, 2, 3, 4],
+        [1, 2, 3, 4],
+        [1, 2, 3, 4],
+        [1, 2, 3, 4],
+        [1.1, 2.2, 3.3, 4.4],
+        [1.1, 2.2, 3.3, 4.4],
+    ]
+    cls.scalar_dtypes = (
+        dtypes.int8,
+        dtypes.int16,
+        dtypes.int32,
+        dtypes.int64,
+        dtypes.uint8,
+        dtypes.uint16,
+        dtypes.uint32,
+        dtypes.uint64,
+        dtypes.float32,
+        dtypes.float64
+    )
+    cls.scalar_shapes = tuple(
+        [tensor_shape.TensorShape([]) for _ in cls.scalar_dtypes])
+
+    cls.list_data = [
+        [[1, 1], [2, 2], [3, 3], [4, 4]],
+        [[1], [2, 2], [3, 3, 3], [4, 4, 4]],
+        [[1, 1], [2, 2], [3, 3], [4, 4]],
+        [[1.1, 1.1], [2.2, 2.2], [3.3, 3.3], [4.4, 4.4]],
+        [[1.1], [2.2, 2.2], [3.3, 3.3, 3.3], [4.4, 4.4, 4.4]],
+        [[1.1, 1.1], [2.2, 2.2], [3.3, 3.3], [4.4, 4.4]],
+    ]
+    cls.list_dtypes = (
+        dtypes.int32,
+        dtypes.int32,
+        dtypes.int64,
+        dtypes.float32,
+        dtypes.float32,
+        dtypes.float64
+    )
+    cls.list_shapes = tuple(
+        [tensor_shape.TensorShape([None]) for _ in cls.list_dtypes])
+
+  def run_test_case(self, dataset, case_data):
+    iterator = dataset.make_one_shot_iterator()
+    next_element = iterator.get_next()
+
+    def is_float(dtype):
+      return dtype == dtypes.float16 or \
+             dtype == dtypes.float32 or \
+             dtype == dtypes.float64
+
+    with self.test_session() as sess:
+      for row in range(len(case_data.data[0])):
+        value = sess.run(next_element)
+        for i, col in enumerate(dataset.columns):
+          if case_data.output_shapes[col].ndims == 0:
+            if is_float(case_data.output_types[col]):
+              self.assertAlmostEqual(value[i], case_data.data[col][row], 4)
```

Comment:
I put 4 because it's really comparing `1 == ~1`, `2 == ~2` etc. from the test data. So from the current test data, we don't require it to be that precise. I can try the default value of 7 digits, but I think it might be too much trouble to try to infer a value and make this function completely generic.

Reply:
Don't worry about this then. Thanks for the efforts!